### ISSUE TITLE
Inconsistent behavior with -isystem vs. -I in CPPFLAGS

### ISSUE THREAD

**Issue by erigler-usgs** [Homebrew/linuxbrew#715]
This is probably not a Linuxbrew bug per se, but perhaps Linuxbrew can fix design flaws in certain brewed packages with little effort. FWIW, everything here was done on a pristine Scientific Linux 7 installation (i.e., a RHEL/CentOS clone).

While brewing Python, everything installed, but as soon as I tried to call pip, I got an immediate "ImportError: No module named zlib". Looking back through brew's build logs, it was evident that zlib.h was not being found (and other stuff, but let's keep this simple for now), and the zlib module was not being compiled. Anyway, I attempted to re-install Python by introducing a tiny change to its Homebrew formula:

```ruby
  cppflags << "-I#{HOMEBREW_PREFIX}/include"
```

Everything seems to work now, but this *should* not have been necessary, since the Linuxbrew build environment sets CPPFLAGS=-isystem/path/to/homebrew/prefix/include. However, in select stages of the complicated Python build process, this option is not getting used properly, and certain packages just refused to see locally installed headers. Presumably this would have never been an issue if I had just installed the zlib-devel package to get the header files in a standard location, but of course this is not just a zlib problem, but a more general issue.

So in short, is there a compelling reason to use '-isystem' instead of '-I' for Linuxbrew's CPPFLAGS?

**Comment by sjackman** [comment]
`-isystem` is used so that header files in the package being compiled are searched before `#{HOMEBREW_PREFIX}/include`.
Python is weird in that its build system parses `CPPFLAGS` looking for directory specified by `-I`, but it ignores directories specified by `-isystem`.
There's a fix over here: https://github.com/Homebrew/linuxbrew/pull/460
And the original issue here: https://github.com/Homebrew/linuxbrew/issues/420

**Comment by sjackman** [comment]
This issue should hopefully be fixed now by cd680d7281f1b59b9603853403f562bd70ac97d2
